src/NuGet.Core/NuGet.Packaging/Signing/Signatures/Signature.cs

```diff
@@ -122,6 +128,37 @@ private static SignatureType GetSignatureType(SignerInfo signer)
             // TODO: Change this to use the new attributes that justin is adding.
             return SignatureType.Author;
         }
+        private static Timestamp GetTimestamp(SignerInfo signer)
+        {
+            var authorUnsignedAttributes = signer.UnsignedAttributes;
+            var timestampCms = new SignedCms();
+
+            foreach (var attribute in authorUnsignedAttributes)
+            {
+                if (string.Equals(attribute.Oid.Value, Oids.SignatureTimeStampTokenAttributeOid))
+                {
+                    timestampCms.Decode(attribute.Values[0].RawData);
+
+                    if (Rfc3161TimestampVerificationUtility.TryReadTSTInfoFromSignedCms(timestampCms, out var tstInfo))
```

Comment:
If this is not true we should throw an exception. Not being able to parse a timestamp when trying to get timestamps should be an error. What do you think @mishra14 ?